components/compositing/constellation.rs

```diff
@@ -384,6 +391,8 @@ impl<LTF: LayoutThreadFactory, STF: ScriptThreadFactory> Constellation<LTF, STF>
             let namespace_id = constellation.next_pipeline_namespace_id();
             PipelineNamespace::install(namespace_id);
             constellation.run();
+            let _ = constellation.resource_thread.send(
+                        net_traits::ControlMsg::SendConstellationMsgChannel(ipc_resource_msg_sender));
```

Comment:
Er. It's disconcerting that the test (presumably) passes right now, since this message will not be sent until Servo is in the process of shutting down - `run` is an endless loop.